core/primitives-core/src/account.rs

```diff
@@ -20,6 +23,173 @@ pub struct Account {
+            Account::AccountV1(acc) => acc.locked,
+        }
+    }
+
+    #[cfg(not(feature = "protocol_feature_add_account_versions"))]
+    pub fn code_hash(&self) -> CryptoHash {
+        self.code_hash
+    }
+
+    #[cfg(feature = "protocol_feature_add_account_versions")]
+    pub fn code_hash(&self) -> CryptoHash {
+        match self {
+            Account::AccountV1(acc) => acc.code_hash,
+        }
+    }
+
+    #[cfg(not(feature = "protocol_feature_add_account_versions"))]
+    pub fn storage_usage(&self) -> StorageUsage {
+        self.storage_usage
+    }
+
+    #[cfg(feature = "protocol_feature_add_account_versions")]
+    pub fn storage_usage(&self) -> StorageUsage {
+        match self {
+            Account::AccountV1(acc) => acc.storage_usage,
+        }
+    }
+
+    #[cfg(not(feature = "protocol_feature_add_account_versions"))]
+    pub fn set_amount(&mut self, amount: Balance) {
+        self.amount = amount;
+    }
+
+    #[cfg(feature = "protocol_feature_add_account_versions")]
+    pub fn set_amount(&mut self, amount: Balance) {
+        match self {
+            Account::AccountV1(acc) => {
+                acc.amount = amount;
+            }
+        }
+    }
+
+    #[cfg(not(feature = "protocol_feature_add_account_versions"))]
+    pub fn set_locked(&mut self, locked: Balance) {
+        self.locked = locked;
+    }
+
+    #[cfg(feature = "protocol_feature_add_account_versions")]
+    pub fn set_locked(&mut self, locked: Balance) {
+        match self {
+            Account::AccountV1(acc) => {
+                acc.locked = locked;
+            }
+        }
+    }
+
+    #[cfg(not(feature = "protocol_feature_add_account_versions"))]
+    pub fn set_code_hash(&mut self, code_hash: CryptoHash) {
+        self.code_hash = code_hash;
+    }
+
+    #[cfg(feature = "protocol_feature_add_account_versions")]
+    pub fn set_code_hash(&mut self, code_hash: CryptoHash) {
+        match self {
+            Account::AccountV1(acc) => {
+                acc.code_hash = code_hash;
+            }
+        }
+    }
+
+    #[cfg(not(feature = "protocol_feature_add_account_versions"))]
+    pub fn set_storage_usage(&mut self, storage_usage: StorageUsage) {
+        self.storage_usage = storage_usage;
+    }
+
+    #[cfg(feature = "protocol_feature_add_account_versions")]
+    pub fn set_storage_usage(&mut self, storage_usage: StorageUsage) {
+        match self {
+            Account::AccountV1(acc) => {
+                acc.storage_usage = storage_usage;
+            }
+        }
+    }
+}
+
+#[cfg(feature = "protocol_feature_add_account_versions")]
+impl BorshDeserialize for Account {
+    fn deserialize(buf: &mut &[u8]) -> Result<Self, Error> {
+        if buf.len() == std::mem::size_of::<AccountV1>() {
+            // This should only ever happen if we have pre-transition account serialized in state
+            // See test_account_size
+            Ok(Account::AccountV1(<AccountV1 as BorshDeserialize>::deserialize(buf)?))
```

Comment:
is `as BorshDeserialize` necessary?

Reply:
Yes, as there is method with same signature in (serde) Serialize